### PR TITLE
Described all router environment variables

### PR DESCRIPTION
**Pull request by knobunc**
Added missing router environment variables to the documentation, and
added descriptons to all of them.

For 3.3 and 3.4.

### CODE REVIEW

**Comment by knobunc** [comment]
@JacobTanenbaum PTAL

File: architecture/core_concepts/routes.adoc [hunk 1]
```diff
@@ -186,21 +186,28 @@ addresses; because of the NAT configuration, the originating IP address
...
-|`*ROUTER_LOG_LEVEL*` | warning
-|`*ROUTER_BACKEND_CHECK_INTERVAL*` | 5000ms
-|`*ROUTER_DEFAULT_CONNECT_TIMEOUT*`| 5s
-|`*ROUTER_DEFAULT_CLIENT_TIMEOUT*`| 30s
-|`*ROUTER_DEFAULT_SERVER_TIMEOUT*`| 30s
-|`*ROUTER_DEFAULT_TUNNEL_TIMEOUT*` | 1h
-|`*ROUTER_SLOWLORIS_TIMEOUT*` | 10s
+|Variable | Default | Description
+|`*ROUTER_SERVICE_HTTP_PORT*` | 80 | What port to listen for http requests on
+|`*ROUTER_SERVICE_HTTPS_PORT*` | 443 | What port to listen for http requests on
```
Comment (JacobTanenbaum) [inline architecture/core_concepts/routes.adoc]: should this be "What port to listen for https requests on"?

Comment (knobunc) [inline architecture/core_concepts/routes.adoc]: It should!

**Comment by JacobTanenbaum** [comment]
LGTM

**Comment by knobunc** [comment]
@adellape PTAL

File: architecture/core_concepts/routes.adoc [hunk 2]
```diff
@@ -186,21 +186,28 @@ addresses; because of the NAT configuration, the originating IP address
...
+|Variable | Default | Description
+|`*ROUTER_SERVICE_HTTP_PORT*` | 80 | What port to listen for http requests on
+|`*ROUTER_SERVICE_HTTPS_PORT*` | 443 | What port to listen for https requests on
+|`*ROUTER_SERVICE_NO_SNI_PORT*` | 10443 | Internal port for some front-end to back-end communication (see note below)
+|`*ROUTER_SERVICE_SNI_PORT*` | 10444 | Internal port for some front-end to back-end communication (see note below)
+|`*ROUTER_SYSLOG_ADDRESS*` |  | Where to send log messages.  Disabled if empty
+|`*ROUTER_LOG_LEVEL*` | warning | What log level to send to the above address
+|`*ROUTER_BACKEND_CHECK_INTERVAL*` | 5000ms | How frequently to check backends for "liveness"
+|`*ROUTER_DEFAULT_CONNECT_TIMEOUT*`| 5s | The maximum connect time
+|`*ROUTER_DEFAULT_CLIENT_TIMEOUT*`| 30s |How long a client has to acknowledge or send data
```
Comment (ramr) [inline architecture/core_concepts/routes.adoc]: Minor nit space between '|' and 'How'.

File: architecture/core_concepts/routes.adoc [hunk 3]
```diff
@@ -186,21 +186,28 @@ addresses; because of the NAT configuration, the originating IP address
...
+|`*ROUTER_DEFAULT_SERVER_TIMEOUT*`| 30s |How long a server has to acknowledge or send data
```
Comment (ramr) [inline architecture/core_concepts/routes.adoc]: Same as above: Minor nit space between '|' and 'How'.

**Comment by knobunc** [comment]
@ramr Updated per your suggestion.  Thanks

**Comment by ramr** [comment]
LGTM

**Comment by zhaozhanqi** [comment]
Could you also help add the following which I knowed:
oc env dc/router TEMPLATE_FILE=/var/lib/haproxy/conf/custom/haproxy-config-custom.template
oc env dc/router RELOAD_INTERVAL=12s

**Comment by knobunc** [comment]
@zhaozhanqi thanks for the suggestion.  By looking for those I found more to add as well.

**Comment by knobunc** [comment]
@openshift/team-documentation this is ready to merge.

**Comment by vikram-redhat** [comment]
@gaurav-nelson - PTAL and if no issues, please merge.

File: architecture/core_concepts/routes.adoc [hunk 4]
```diff
@@ -186,21 +186,46 @@ addresses; because of the NAT configuration, the originating IP address
...
+|`*DEFAULT_CERTIFICATE*` |  | The contents of a default certificate to use for routes that don't expose a TLS server cert; in PEM format.
+|`*DEFAULT_CERTIFICATE_DIR*` |  | A path to a directory that contains a file named tls.crt. If tls.crt is not a PEM file which also contains a private key, it is first combined with a file named tls.key in the same directory. The PEM-format contents are then used as the default certificate. Only used if DEFAULT_CERTIFICATE or DEFAULT_CERTIFICATE_PATH are not specified.
+|`*DEFAULT_CERTIFICATE_PATH*` |  | A path to default certificate to use for routes that don't expose a TLS server cert; in PEM format. Only used if DEFAULT_CERTIFICATE is not specified.
+|`*EXTENDED_VALIDATION*` | true | Whether to perform an additional extended validation step on all routes admitted by this router.
+|`*NAMESPACE_LABELS*` |  | A label selector to apply to namespaces to watch, empty means all.
+|`*PROJECT_LABELS*` |  | A label selector to apply to projects to watch, emtpy means all.
+|`*RELOAD_SCRIPT*` |  | The path to the reload script to use to reload the router.
+|`*ROUTER_BACKEND_CHECK_INTERVAL*` | 5000ms | How frequently to check backends for "liveness".
```
Comment (gaurav-nelson) [inline architecture/core_concepts/routes.adoc]: Can we please change the description from "**How frequently to....**" to **Time interval between subsequent "liveness" checks on backends.**

File: architecture/core_concepts/routes.adoc [hunk 5]
```diff
@@ -186,21 +186,46 @@ addresses; because of the NAT configuration, the originating IP address
...
+|`*ROUTER_COMPRESSION_MIME*` | "text/html text/plain text/css" | A space separated list of mime types to compress.
+|`*ROUTER_DEFAULT_CLIENT_TIMEOUT*`| 30s | How long a client has to acknowledge or send data.
```
Comment (gaurav-nelson) [inline architecture/core_concepts/routes.adoc]: Suggestion: Please change "_How long a client has to acknowledge or send data._" to "**Time interval within which a client has to acknowledge or send data.**"

File: architecture/core_concepts/routes.adoc [hunk 6]
```diff
@@ -186,21 +186,46 @@ addresses; because of the NAT configuration, the originating IP address
...
+|`*ROUTER_DEFAULT_CONNECT_TIMEOUT*`| 5s | The maximum connect time.
+|`*ROUTER_DEFAULT_SERVER_TIMEOUT*`| 30s | How long a server has to acknowledge or send data.
```
Comment (gaurav-nelson) [inline architecture/core_concepts/routes.adoc]: Suggestion: Please change "How long a server has to acknowledge or send data." to "Time interval within which a server has to acknowledge or send data."

File: architecture/core_concepts/routes.adoc [hunk 7]
```diff
@@ -186,21 +186,46 @@ addresses; because of the NAT configuration, the originating IP address
...
+|`*ROUTER_DEFAULT_TUNNEL_TIMEOUT*` | 1h | How long to allow TCP or WebSocket connections to remain open.
```
Comment (gaurav-nelson) [inline architecture/core_concepts/routes.adoc]: Suggestion: Please change "How long to allow TCP or WebSocket connections to remain open." to "Time interval to allow TCP or WebSocket connections to remain open."  OR "Time interval after which TCP or WebSocket connections will close."

File: architecture/core_concepts/routes.adoc [hunk 8]
```diff
@@ -186,21 +186,46 @@ addresses; because of the NAT configuration, the originating IP address
...
+|`*ROUTER_ENABLE_COMPRESSION*`| false | If "true", compress responses when possible.
+|`*ROUTER_LOG_LEVEL*` | warning | What log level to send to the above address.
```
Comment (gaurav-nelson) [inline architecture/core_concepts/routes.adoc]: Suggestion: Please change "What log level to send to the above address." to "The log level to send to the specified address." (OR "The log level to send to the above address." above meaning is not clear though.)

File: architecture/core_concepts/routes.adoc [hunk 9]
```diff
@@ -186,21 +186,46 @@ addresses; because of the NAT configuration, the originating IP address
...
+|`*ROUTER_OVERRIDE_HOSTNAME*`|  | If set, override the spec.host value for a route with the template in ROUTER_SUBDOMAIN.
+|`*ROUTER_SERVICE_HTTPS_PORT*` | 443 | What port to listen for https requests on.
```
Comment (gaurav-nelson) [inline architecture/core_concepts/routes.adoc]: Suggestion: Please change "What port to listen for https requests on." to "Port to listen for https requests." OR "Port on which to listen for https requests."

File: architecture/core_concepts/routes.adoc [hunk 10]
```diff
@@ -186,21 +186,46 @@ addresses; because of the NAT configuration, the originating IP address
...
+|`*ROUTER_SERVICE_HTTP_PORT*` | 80 | What port to listen for http requests on.
```
Comment (gaurav-nelson) [inline architecture/core_concepts/routes.adoc]: Suggestion: Please change "What port to listen for http requests on." to "Port to listen for http requests." OR "Port on which to listen for http requests."

**Comment by gaurav-nelson** [comment]
Everything looks good. I will do a merge and then update the language changes I suggested.

**Comment by knobunc** [comment]
Thank you @gaurav-nelson.  Your suggestions look good.

If this can go in 3.4 and 3.3 that would be nice, but really getting it in 3.4 alone is sufficient.

**Comment by adellape** [comment]
@knobunc Can you also squash down to 1 commit? Will leave the merging to @gaurav-nelson though here.

**Comment by knobunc** [comment]
Squashed and addressed @gaurav-nelson's comments.

**Comment by gaurav-nelson** [comment]
Follow up is now in PR #3251 (https://github.com/openshift/openshift-docs/pull/3251)

**Comment by gaurav-nelson** [comment]
@knobunc Can you please confirm that all these options available in 3.3 now? Before it gets released this Monday. Thanks.

**Comment by knobunc** [comment]
@gaurav-nelson thank you for checking.  I just ran through them all and there are two that are not:
 - ROUTER_ENABLE_COMPRESSION
 - ROUTER_COMPRESSION_MIME

**Pull request merged**